src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -1792,6 +1821,150 @@ export class ResultsViewPageStore {
+                await internalClient.fetchMutationEnrichmentsUsingPOST({
+                    molecularProfileId: this.selectedEnrichmentMutationProfile.molecularProfileId,
+                    enrichmentType: "SAMPLE",
+                    enrichmentFilter: {alteredIds: this.alteredSamples.result.map(s => s.sampleId),
+                        unalteredIds: this.unalteredSamples.result.map(s => s.sampleId)}})) : [];
+        }
+    });
+
+    readonly copyNumberEnrichmentProfiles = remoteData<MolecularProfile[]>({
+        await: () => [
+            this.molecularProfilesInStudies
+        ],
+        invoke: async () => {
+            const profiles: MolecularProfile[] = _.filter(this.molecularProfilesInStudies.result,
+                (profile: MolecularProfile) => profile.molecularAlterationType === AlterationTypeConstants.COPY_NUMBER_ALTERATION
+                    && profile.datatype === "DISCRETE");
+            this.selectedEnrichmentCopyNumberProfile = profiles[0];
+            return profiles;
+        }
+    });
+
+    readonly copyNumberHomdelEnrichmentData = remoteData<AlterationEnrichment[]>({
+        await: () => [
+            this.alteredSamples,
+            this.unalteredSamples,
+            this.copyNumberEnrichmentProfiles
+        ],
+        invoke: async () => {
+            return this.selectedEnrichmentCopyNumberProfile ? this.getCopyNumberEnrichmentData(this.alteredSamples.result, 
+                this.unalteredSamples.result, "HOMDEL") : [];
+        }
+    });
+
+    readonly copyNumberAmpEnrichmentData = remoteData<AlterationEnrichment[]>({
+        await: () => [
+            this.alteredSamples,
+            this.unalteredSamples,
+            this.copyNumberEnrichmentProfiles
+        ],
+        invoke: async () => {
+            return this.selectedEnrichmentCopyNumberProfile ? this.getCopyNumberEnrichmentData(this.alteredSamples.result, 
+                this.unalteredSamples.result, "AMP") : [];
+        }
+    });
+
+    private async getCopyNumberEnrichmentData(alteredSamples: Sample[], unalteredSamples: Sample[], 
+        copyNumberEventType: "HOMDEL" | "AMP"): Promise<AlterationEnrichment[]> {
+        
+        return this.sortEnrichmentData(await internalClient.fetchCopyNumberEnrichmentsUsingPOST({
+            molecularProfileId: this.selectedEnrichmentCopyNumberProfile.molecularProfileId,
+            copyNumberEventType: copyNumberEventType,
+            enrichmentType: "SAMPLE",
+            enrichmentFilter: {
+                alteredIds: alteredSamples.map(s => s.sampleId),
+                unalteredIds: unalteredSamples.map(s => s.sampleId)
+        }}));
+    }
+
+    readonly mRNAEnrichmentProfiles = remoteData<MolecularProfile[]>({
+        await: () => [
+            this.molecularProfilesInStudies
+        ],
+        invoke: async () => {
+            let profiles: MolecularProfile[] = _.filter(this.molecularProfilesInStudies.result,
```

Comment:
not a big deal but couldn't the filter and _remote steps be done in the filter?

Reply:
_.remove is actually used to move the removed elements at the top of the list, so not a permanent _.remove operation. I'll add a comment.